gobblin-cluster/src/main/java/gobblin/cluster/ScheduledJobConfigurationManager.java

```diff
@@ -0,0 +1,131 @@
+
+  private static final long DEFAULT_JOB_SPEC_REFRESH_INTERVAL = 60;
+
+  private Map<URI, JobSpec> jobSpecs;
+
+  private final long refreshIntervalInSeconds;
+
+  private final ScheduledExecutorService fetchJobSpecExecutor;
+
+  private final SpecExecutorInstanceConsumer specExecutorInstanceConsumer;
+
+  private final ClassAliasResolver<SpecExecutorInstanceConsumer> aliasResolver;
+
+  public ScheduledJobConfigurationManager(EventBus eventBus, Config config) {
+    super(eventBus, config);
+
+    this.jobSpecs = Maps.newHashMap();
+    if (config.hasPath(GobblinClusterConfigurationKeys.JOB_SPEC_REFRESH_INTERVAL)) {
+      this.refreshIntervalInSeconds = config.getDuration(GobblinClusterConfigurationKeys.JOB_SPEC_REFRESH_INTERVAL,
+          TimeUnit.SECONDS);
+    } else {
+      this.refreshIntervalInSeconds = DEFAULT_JOB_SPEC_REFRESH_INTERVAL;
+    }
+
+    this.fetchJobSpecExecutor = Executors.newSingleThreadScheduledExecutor(
+        ExecutorsUtils.newThreadFactory(Optional.of(LOGGER), Optional.of("FetchJobSpecExecutor")));
+
+    this.aliasResolver = new ClassAliasResolver<>(SpecExecutorInstanceConsumer.class);
+    try {
+      String specExecutorInstanceConsumerClassName = GobblinClusterConfigurationKeys.DEFAULT_SPEC_EXECUTOR_INSTANCE_CONSUMER_CLASS;
+      if (config.hasPath(GobblinClusterConfigurationKeys.SPEC_EXECUTOR_INSTANCE_CONSUMER_CLASS_KEY)) {
+        specExecutorInstanceConsumerClassName = config.getString(GobblinClusterConfigurationKeys.SPEC_EXECUTOR_INSTANCE_CONSUMER_CLASS_KEY);
+      }
+      LOGGER.info("Using SpecExecutorInstanceConsumer ClassNameclass name/alias " + specExecutorInstanceConsumerClassName);
+      this.specExecutorInstanceConsumer = (SpecExecutorInstanceConsumer) ConstructorUtils
+          .invokeConstructor(Class.forName(this.aliasResolver.resolve( specExecutorInstanceConsumerClassName)), config);
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException
+          | ClassNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    LOGGER.info("Starting the " + ScheduledJobConfigurationManager.class.getSimpleName());
+
+    LOGGER.info(String.format("Scheduling the job spec refresh task with an interval of %d second(s)",
+        this.refreshIntervalInSeconds));
+
+    // Schedule the job config fetch task
+    this.fetchJobSpecExecutor.scheduleAtFixedRate(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          fetchJobSpecs();
+        } catch (InterruptedException | ExecutionException e) {
+          LOGGER.error("Failed to fetch job specs", e);
+          throw new RuntimeException("Failed to fetch specs", e);
+        }
+      }
+    }, 0, this.refreshIntervalInSeconds, TimeUnit.SECONDS);
+  }
+
+  private void fetchJobSpecs() throws ExecutionException, InterruptedException {
```

Comment:
We will need a mechanism to persist the job specs in the cluster or request they be re-sent from GaaS since the messages can't stay in Kafka indefinitely.